zfit/models/cache.py

```diff
@@ -0,0 +1,137 @@
+from zfit import supports
+from zfit.models.functor import BaseFunctor
+
+import tensorflow as tf
+import zfit.z.numpy as znp
+
+
+def get_value(cache: tf.Variable, flag: tf.Variable, func):
+    def autoset_func():
+        val = func()
+        cache.assign(val, read_value=False)
+        return cache
+
+    def use_cache():
+        return cache
+
+    val = tf.cond(flag, use_cache, autoset_func)
+    return val
+
+
+class CacheablePDF(BaseFunctor):
+    def __init__(self, pdf, cache_tolerance=None, **kwargs):
```

Comment:
This needs documentation. On an example, see another PDF. It should describe the behavior, arguments of the PDF